### PR TITLE
Review and Update Developing Poly doc

### PR DESCRIPTION
**Pull request by lread**
Re-order and re-section some content for an easier to follow flow.

Add a Contributing section.
Currently my (@lread) preferences and biases, polylith core team can adapt to their point of view.

Add a few more details on cljdoc.

Give concrete example of using other workspace usage.

Edits for clarity.

### CODE REVIEW

**Comment by lread** [comment]
@tengstrand, ready for your review and feedback!
As always, please read the commit message before reviewing.

File: doc/developing-poly.adoc [hunk 1]
```diff
@@ -1,102 +1,136 @@
...
 [source,shell]
 ----
 poly info fake-tag:
 ----
 
-=== fake-poly
+=== `fake-poly`
 
-This argument is used to fake that we are using a non-snapshot version of the `poly` tool, and not the `polyx` tool.
+Fake that you are using a stable production release of `poly` when you might actually be using a `SNAPSHOT` release, ' polyx`, or both.
```
Comment (tengstrand) [inline doc/developing-poly.adoc]: Change ' to ` for polyx

File: doc/developing-poly.adoc [hunk 2]
```diff
@@ -1,102 +1,136 @@
...
 
 [source,shell]
 ----
-# Excludes "-SNAPSHOT" from the version at the top of the help output:
-poly help :fake-poly
-# Excludes "-SNAPSHOT" from the Polylith ascii art logo:
-poly :fake-poly
+poly help :fake-polya # <1>
```
Comment (tengstrand) [inline doc/developing-poly.adoc]: Should be :fake-poly I guess.

File: doc/developing-poly.adoc [hunk 3]
```diff
@@ -145,58 +237,66 @@ Among other https://asciidoctor.org/docs/asciidoc-recommended-practices/#one-sen
 * For command-line `poly create workspace top-ns:my-ns branch:my-branch :commit` refer to:
 ** `create workspace` as a command
 ** `top-ns:my-ns`, `branch:my-branch` and `:commit` as arguments
-* Always link to referenced commands and concepts to make it easier for readers to explore and dig into details.
+* Link to referenced commands and concepts to make it easier for readers to explore and dig into details.
 * Use "we" when talking from the Polylith core team and "you" when directing the reader through some steps or a tutorial.
```
Comment (tengstrand) [inline doc/developing-poly.adoc:246]: Here somewhere in this list we could tell people to use special syntax for keyboard shortcuts, found [here](https://cljdoc.org/d/polylith/clj-poly/0.2.18-SNAPSHOT/doc/shell#quit).

File: doc/developing-poly.adoc [hunk 4]
```diff
@@ -1,102 +1,136 @@
...
+We expect you to include/update tests for your change.
+
+You've probably gotten the sense that providing good docs for our users is important to us.
+When your change affects our docs or needs to be described for our users, we expect you to do so.
+
+=== Submitting a Pull Request
+
+Please never force push on your PR, as this makes reviewing incremental changes impossible for us.
+When we merge your PR, we'll usually squash it, so that will clean up any rambling work-in-progress.
+
```
Comment (tengstrand) [inline doc/developing-poly.adoc:34]: Tell people to add a row in [next-release.txt](https://github.com/polyfy/polylith/blob/master/next-release.txt) when they create a PR. Additional info could be that this file is cleared when we make a final/stable release.

**Review by tengstrand — CHANGES_REQUESTED** [review]
I really like your changes. Awesome job!

File: doc/developing-poly.adoc (hunk 1 above)
Comment (lread) [reply doc/developing-poly.adoc]: tx!

File: doc/developing-poly.adoc (hunk 2 above)
Comment (lread) [reply doc/developing-poly.adoc]: tx!

File: doc/developing-poly.adoc (hunk 3 above)
Comment (lread) [reply doc/developing-poly.adoc:246]: will do

File: doc/developing-poly.adoc (hunk 4 above)
Comment (lread) [reply doc/developing-poly.adoc:34]: yep, can do

File: doc/developing-poly.adoc (hunk 1 above)
Comment (lread) [reply doc/developing-poly.adoc]: fixed

File: doc/developing-poly.adoc (hunk 2 above)
Comment (lread) [reply doc/developing-poly.adoc]: vi keys get me sometimes, fixed!

File: doc/developing-poly.adoc (hunk 3 above)
Comment (lread) [reply doc/developing-poly.adoc:246]: done!

File: doc/developing-poly.adoc (hunk 4 above)
Comment (lread) [reply doc/developing-poly.adoc:34]: Added note

**Comment by lread** [comment]
@tengstrand, thanks, all review feedback addressed!

**Review by tengstrand — APPROVED** [review]

**Pull request merged**